### PR TITLE
feat: standardize make extract_translations

### PR DESCRIPTION
**Pull request by shadinaif**
feat: standardize make extract_translations

This PR prepares the repository to comply with [openedx-translations](https://github.com/openedx/openedx-translations) by doing the following:

* Create translation directory `conf/locale`. This is the default used one in [openedx-translations](https://github.com/openedx/openedx-translations)
* Remove `locale` directory. It's not actually used. It was added when `completion` is stripped out of `edx-platform`, there is no local translation, and there is no translation project named `completion` in transifex
* Update `make extract_translations` to use `django-admin` instead of `manage.py`

- [x] Tested on local devstack palm to ensure that everything renders fine
- [x] ~~Tested on local devstack palm to ensure that local translation still works~~ no local translations exist in this repo

References
----------

This pull request is part of the [FC-0012 project](https://openedx.atlassian.net/l/cp/XGS0iCcQ) which is sparked by the [Translation Infrastructure update OEP-58](https://open-edx-proposals.readthedocs.io/en/latest/architectural-decisions/oep-0058-arch-translations-management.html#specification).

Check the links above for full information about the overall project.

Internalization is being rearchitected in Open edX Python, XBlock, Micro-frontend, and other projects. There are a number of immediately visible changes:
 - Remove source and language translations from the repositories, hence no `.json`, `.po` or `.mo` files will be committed into the repos.
 - Add standardized `make extract_translations` in all repositories
 - Push user-facing messages strings into [openedx/openedx-translations](https://github.com/openedx/openedx-translations/).
 - Integrate root repositories with [openedx/openedx-atlas](https://github.com/openedx/openedx-atlas/) to pull translations on build/deploy time

Breaking Changes
----------------

One of the primary goals of the project is to **avoid breaking changes**. If you noticed any suspicious code, please raise your concern. But before that, please know the strategy we're following to avoid breaking changes

**For XBlocks:**
 - The standard translation path must be `conf/locale`. Therefore, we are creating a link from `conf/locale` to `translations` so Atlas can find the path without disrupting the current way of having translations locally within the XBlocks
 - [openedx-translations](https://github.com/openedx/openedx-translations) will have a related PR that adds the XBlock to the pipeline. This will not affect the current way of managing/updating translations
 - At the end of the project, a clear change log will be added and all translations will be handled by Atlas. Thus, the local translation will be removed from the repo within the version bump
 - A notification for the community will be published to ensure that everyone knows why translations directories are removed from all repos

### CODE REVIEW

**Comment by shadinaif** [comment]
this one is ready @OmarIthawi, @brian-smith-tcril

**Comment by shadinaif** [comment]
@OmarIthawi , replacing `manage.py` with `django-admin` is to avoid falling into the need for a `testing_settings.py`, that's all

**Comment by OmarIthawi** [comment]
> @OmarIthawi , replacing `manage.py` with `django-admin` is to avoid falling into the need for `testing_settings.py`, that's all

Thanks for explaining that. I wasn't aware.

File: locale/config.yaml
Comment (brian-smith-tcril) [inline locale/config.yaml:1]: could you add something to the PR comment explaining why removing all of these is ok?

Comment (shadinaif) [reply locale/config.yaml:1]: @brian-smith-tcril  I've added the explanation to the PR description:

It's not actually used. It was added when completion is stripped out of `edx-platform`, there is no local translation, and there is no translation project named `completion` in transifex

**Review by brian-smith-tcril — APPROVED** [review]

**Comment by brian-smith-tcril** [comment]
@mphilbrick211 this one is good to go! The spreadsheet says "No extra review required" so it should just be a matter of getting a CC with perms on this repo to hit the merge button.

**Comment by mphilbrick211** [comment]
Hi @chennighan2u and @openedx/masters-devs-cosmonauts! Would someone be able to review and merge this for us? Thanks!

**Comment by openedx-webhooks** [comment]
@shadinaif 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/completion&entry.867055334=https://github.com/openedx/completion/pull/232&entry.1484655318=https://github.com/shadinaif&entry.752974735=2023-04-26+14:44&entry.1917517419=2023-05-09+14:19&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**